pandas/core/generic.py

```diff
@@ -2341,6 +2341,7 @@ def to_json(
         default_handler: Callable[[Any], JSONSerializable] | None = None,
         lines: bool_t = False,
         compression: CompressionOptions = "infer",
+        mode: str = "w",
```

Comment:
Is mode needed? I think we expect/require that file handles are opened in binary mode when the user request compression.

Reply:
Nope, not needed. I realised that the passthrough described in

https://github.com/pandas-dev/pandas/pull/44787/files#diff-132ee3be1f83a9f885442f45ed9ccbc96796ae28f97991b7c99ce25d44fd6af7R206

doesn't yet work. Fixed in https://github.com/pandas-dev/pandas/pull/44787/commits/57eba0ad7b63c984fb69c8c2a7e1a40ae3721bd6, and removed the added `mode` parameter.